File: src/platform/audit-service/src/main/java/com/enterprise/auditservice/controller/HealthController.java
package com.enterprise.auditservice.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

/**
 * Lightweight liveness probe endpoint.
 *
 * <p>Returns HTTP 200 with {@code {"status":"ok"}} whenever the application
 * context is up.  Kubernetes liveness and readiness probes, as well as the
 * platform health-check-service, poll this endpoint.</p>
 */
@RestController
public class HealthController {

    /**
     * Liveness probe.
     *
     * @return 200 OK with body {@code {"status":"ok"}}
     */
    @GetMapping("/healthz")
    public ResponseEntity<Map<String, String>> healthz() {
        return ResponseEntity.ok(Map.of("status", "ok"));
    }
}

File: src/commerce/order-service/src/main/kotlin/com/shopos/orderservice/controller/HealthController.kt
package com.shopos.orderservice.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class HealthController {

    @GetMapping("/healthz")
    fun health(): Map<String, String> = mapOf("status" to "ok")
}
